app/controllers/concerns/foreman/controller/provisioning_templates.rb

```diff
@@ -42,4 +43,22 @@ def process_template_kind
   def type_name_singular
     @type_name_singular ||= resource_class.to_s.underscore
   end
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:allowed_registration_vars)
+                               .flatten.compact.uniq
+
+    organization = Organization.authorized(:view_organizations).find(params['organization_id']) if params['organization_id'].present?
+    location = Location.authorized(:view_locations).find(params['location_id']) if params['location_id'].present?
+    host_group = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_id']) if params["hostgroup_id"].present?
```

Comment:
For my understanding: if find fails, does it raise an exception or return nil? IMHO it should raise an error if the requested ID is not found.

Reply:
That's current behavior - if requested ID is not found you will get an error.

Reply:
semi related - https://projects.theforeman.org/issues/30696

Reply:
Can you capture that in a test? It'd be good to capture that requirement as code.

Reply:
@ekohl I will do it (the tests) as a part of https://projects.theforeman.org/issues/30696, if you okay with it.

Reply:
I'm always a fan of adding tests as part of the commit itself. That makes it much easier in the future to see where it came from. Otherwise you make git blame much harder. I'm ok with leaving out functionality, but tests belong with the actual change.